internal/service/certificatesigningrequest.go

```diff
@@ -265,7 +269,8 @@ func (h *ServiceHandler) ReplaceCertificateSigningRequest(ctx context.Context, r
 			approveResp, _ := h.ApproveCertificateSigningRequest(ctx, approveReq)
 			_, ok := approveResp.(server.ApproveCertificateSigningRequest200JSONResponse)
 			if !ok {
-				return server.ReplaceCertificateSigningRequest400JSONResponse{Message: "CSR created but could not be approved"}, nil
+				msg := fmt.Sprintf("enrollment CSR for %s could not be auto-approved: %s", request.Name, approveResp)
+				return server.ReplaceCertificateSigningRequest400JSONResponse{Message: msg}, nil
```

Comment:
A HTTP 400 (Bad Request) error is for when the API request isn't valid, not for when there are errors during handling of the request.

Reply:
We could grow a 500 error here, but the API currently only has 200, 201, 400, 401, 404, and 409.

Reply:
@fzdarsky This is already a 400 error so that is not a new thing I am introducing in this PR. I am planning to submit a separate PR that updates the error codes where needed. Can this be merged as-is?

Reply:
Sure. The PR removes the length requirement, so that'll be helpful. It's mainly the change here that'll need to be undone in a follow-up.